--- a/Filters/ParallelDIY2/vtkRedistributeDataSetFilter.cxx	2022-03-08 00:35:31.589443296 -0800
+++ b/Filters/ParallelDIY2/vtkRedistributeDataSetFilter.cxx.patched	2022-03-08 00:36:27.214917422 -0800
@@ -518,6 +518,7 @@
   {
     assert(vtkUnstructuredGrid::SafeDownCast(outputDO) != nullptr);
     vtkNew<vtkAppendFilter> appender;
+    appender->MergePointsOn();
 
     using Opts = vtk::DataObjectTreeOptions;
     for (vtkDataObject* part : vtk::Range(result.GetPointer(),
