--- a/ThirdParty/diy2/vtkdiy2/include/vtkdiy2/serialization.hpp	2022-01-28 20:11:13.058442761 +0000
+++ b/ThirdParty/diy2/vtkdiy2/include/vtkdiy2/serialization_patched.hpp	2022-01-28 20:11:54.151074423 +0000
@@ -92,11 +92,6 @@
   template<class T>
   struct Serialization: public detail::Default
   {
-#if (defined(__clang__) && !defined(__ppc64__)) || (defined(__GNUC__) && __GNUC__ >= 5)
-    //exempt power-pc clang variants due to: https://gitlab.kitware.com/vtk/vtk-m/issues/201
-    static_assert(std::is_trivially_copyable<T>::value, "Default serialization works only for trivially copyable types");
-#endif
-
     static void         save(BinaryBuffer& bb, const T& x)          { bb.save_binary((const char*)  &x, sizeof(T)); }
     static void         load(BinaryBuffer& bb, T& x)                { bb.load_binary((char*)        &x, sizeof(T)); }
     static size_t       size(const T& x)                            { return sizeof(T); }
